File: 3rd Semester/Advanced Programming Methods(Done)/Laboratory/src (BUX)/Model/Utils/MyDictionary.java
package Model.Utils;
import Exceptions.MyDictionaryException;

import java.util.*;

public class MyDictionary<K,V> implements MyIDictionary<K,V>{

    private Map<K,V> map;
    public MyDictionary()
    {
        map=new HashMap<K,V>();
    }

    public Map<K, V> getMap() {
        return map;
    }

    public void setMap(Map<K, V> map) {
        this.map = map;
    }

    public void put(K key, V val){
        map.put(key,val);
    }

    public String toString(){
        String msg = "";
        for(K name:map.keySet())
        {
            String key = name.toString();
            msg += key + "-->" + map.get(key).toString()+";";
        }
        return msg;
    }
    public V lookup(K key) throws MyDictionaryException{
        if(map.get(key)!=null)
            return map.get(key);
        else
            throw new MyDictionaryException("Nonexistent value!");
    }

    public int isDefined(K key){
        if (map.get(key)!=null)
            return 1;
        return 0;
    }

    public void update(K key, V val){
        map.put(key,val);
    }

    public void print() {

        for(K name:map.keySet()) {
            String key = name.toString();
            System.out.println(key + " " + map.get(key));
        }

    }

    public Map<K,V> getContent(){return map;}
}

File: 3rd Semester/Advanced Programming Methods/Laboratory/Toy Interpreter C#/sharplab/Model/Utils/MyDictionary.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace sharplab.Model.Utils
{
    class MyDictionary<K,V> : IMyDictionary<K,V>
    {
        private SortedDictionary<K,V> dict;

        public MyDictionary()
        {
            dict = new SortedDictionary<K,V>();
        }


        public SortedDictionary<K,V> GetDictionary()
        {
            return dict; 
        }



        public int IsDefined(K key)
        {
            if( dict.ContainsKey(key))
                return 1;
            return 0;
        }



        public V LookUp(K key)
        {
            if (this.IsDefined(key) != 0)
                return dict.GetValueOrDefault(key);
            else
                throw new Exception("Nonexistent value!");
        }



        public override string ToString()
        {
            string s = "";
            var arrayOfAllKeys = dict.Keys.ToArray();
            var arrayOfAllValues = dict.Values.ToArray();
            for (int i = 0; i < arrayOfAllKeys.Length; i++)
            {
                s += arrayOfAllKeys[i].ToString();
                s += "->";
                s += arrayOfAllValues[i].ToString();
                s += ";";
            }
            return s;
        }



        public void Put(K key, V value)
        {
            dict.Add(key, value);
        }



        public void Update(K key, V value)
        {
            dict[key] = value;
        }

    }

    
}
